core/arch/arm/kernel/secure_partition.c

```diff
@@ -68,6 +85,31 @@ struct sp_session *sp_get_session(uint32_t session_id)
 	return NULL;
 }
 
+TEE_Result sp_get_partitions_info(struct ffa_partition_info *fpi,
+				  size_t *elements)
+{
+	size_t in_elements = *elements;
+	struct sp_session *s = NULL;
+	uint32_t count = 0;
+
+	TAILQ_FOREACH(s, &open_sp_sessions, link) {
+		if (s->state != sp_dead)
+			continue;
+		count++;
+		if (count < in_elements) {
+			spmc_fill_partition_entry(fpi, s->endpoint_id, 1);
+			fpi++;
+		}
+	}
+
+	if (count >= in_elements)
+		return TEE_ERROR_SHORT_BUFFER;
+
+	*elements = count;
```

Comment:
This should be done before the return above.

Reply:
Done